src/Processors/Merges/Algorithms/GraphiteRollupSortedAlgorithm.cpp

```diff
@@ -298,8 +304,8 @@ void GraphiteRollupSortedAlgorithm::GraphiteRollupMergedData::accumulateRow(RowR
 
 GraphiteRollupSortedAlgorithm::GraphiteRollupMergedData::~GraphiteRollupMergedData()
 {
-    if (aggregate_state_created)
-        std::get<1>(current_rule)->function->destroy(place_for_aggregate_state.data());
+    if (const Graphite::AggregationPattern * aggregation_pattern = extract_aggregation_pattern_function(current_rule))
+        aggregation_pattern->function->destroy(place_for_aggregate_state.data());
```

Comment:
Instead of rewriting to `std::optional`, will it not be sufficient to add a check here (dtor) whether `aggregation_pattern->function` is `nullptr`?

Reply:
yes, that could be one approach but it still depends on the aggregate_state_created boolean variable. I just wanted to get rid of it and replace with std::optional.